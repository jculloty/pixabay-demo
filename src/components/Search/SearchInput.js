import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class SearchInput extends PureComponent {
  static propTypes = {
    searchText: PropTypes.string.isRequired,
    searchTextChanged: PropTypes.func.isRequired,
    onEnter: PropTypes.func.isRequired,
  };
  domElement = React.createRef();

  componentDidMount() {
    if (this.textInput) {
      this.textInput.focus();
    }
  }

  setTextInputRef = (element) => {
    this.textInput = element;
  };

  handleKeyPress = (event) => {
    if (event.key === 'Enter') {
      this.props.onEnter();
    }
  }

  handleChange = (event) => {
    this.props.searchTextChanged(event.target.value);
  }

  render() {
    return <input type="text"
      placeholder="Search images, vectors and photos"
      value={this.props.searchText}
      onChange={this.handleChange}
      ref={this.setTextInputRef}
      onKeyPress={this.handleKeyPress}
    />;
  }
}

export default SearchInput;
