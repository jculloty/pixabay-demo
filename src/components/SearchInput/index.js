import React from 'react';
import PropTypes from 'prop-types';

function SearchInput(props) {
  return <input type="text" placeholder="Search images, vectors and photos" value={props.searchText} onChange={props.searchTextChanged} />;
}

SearchInput.propTypes = {
  searchText: PropTypes.string.isRequired,
  searchTextChanged: PropTypes.func.isRequired,
};

export default SearchInput;
