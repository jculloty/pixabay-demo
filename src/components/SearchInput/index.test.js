import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import SearchInput from '.';

Enzyme.configure({ adapter: new Adapter() });

const buildComponent = function(text = '', onChange = () => {}) {
  return <SearchInput searchText={text} searchTextChanged={onChange} />;
};

describe('SearchInput', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(buildComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an input', () => {
    const wrapper = shallow(buildComponent());
    expect(wrapper.find('input').exists()).toEqual(true);
  });

  it('searchText is passed to the rendered input', () => {
    const searchText = 'testing';
    const wrapper = shallow(buildComponent(searchText));
    expect(wrapper.find(`input[value="${searchText}"]`).exists()).toEqual(true);
  });

  it('Changes trigger the correct method', () => {
    const changeMock = jest.fn();
    const changedText = 'testing';

    const wrapper = mount(buildComponent('', changeMock));
    const input = wrapper.find('input');

    input.simulate('change', { target: { value: changedText } });

    expect(changeMock.mock.calls.length).toBe(1);
  });
});
