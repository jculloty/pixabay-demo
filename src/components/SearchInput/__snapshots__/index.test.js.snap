// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<input
  onChange={[Function]}
  placeholder="Search images, vectors and photos"
  type="text"
  value=""
/>
`;
